File: chitchat/src/main/java/com/group/chitchat/model/dto/UserForEditDto.java
package com.group.chitchat.model.dto;

import com.group.chitchat.model.enums.Gender;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UserForEditDto {

  private String firstname;
  private String lastname;
  private String role;
  private String avatar;
  private String nativeLanguage;
  private LocalDate dob;
  private Gender gender;

  @Override
  public String toString() {
    return "UserForEditDto{"
        + "firstname='" + firstname + '\''
        + ", lastname='" + lastname + '\''
        + ", role='" + role + '\''
        + ", avatar='" + avatar + '\''
        + ", nativeLanguage='" + nativeLanguage + '\''
        + ", dob=" + dob
        + ", gender=" + gender
        + '}';
  }
}

File: front/src/main/angular/src/app/model/UserForEditDto.ts
import {Gender} from "./Gender";

export class UserForEditDto {
  firstname: string;
  lastname: string;
  role: string;
  avatar: string;
  nativeLanguage: string;
  dob: string;
  gender: Gender;

  constructor(firstname: string, lastname: string, role: string, avatar: string,
              nativeLanguage: string, dob: string, gender: Gender) {
    this.firstname = firstname;
    this.lastname = lastname;
    this.role = role;
    this.avatar = avatar;
    this.nativeLanguage = nativeLanguage;
    this.dob = dob;
    this.gender = gender;
  }
}